image/image_linux.go

```diff
@@ -392,6 +392,18 @@ func (s *imageSeeder) start(db *asserts.Database, optSnaps []*seedwriter.Options
 	return s.w.Start(db, newFetcher)
 }
 
+func (s *imageSeeder) validateSnapArchs(snaps []*seedwriter.SeedSnap) error {
+	for _, sn := range snaps {
+		for _, a := range sn.Info.Architectures {
+			if a != "all" && a != s.model.Architecture() {
```

Comment:
model.Architecture() might be unset as its optional in the case of old-style classic models, so we need to do something about the case where model.Architecture() is ""

Reply:
In case the architecture is unset, does this mean 'the architecture of the host'? or how would I deal with this?

Reply:
judging from the code that configures the ToolingStore the fallback is first opts.Architecture and if that is also not set then it becomes the host arch (arch.DpkgArchitecture())

Reply:
Thank you, changed this!